include/swift/Runtime/Metadata.h

```diff
@@ -590,20 +590,22 @@ swift_relocateClassMetadata(const ClassDescriptor *descriptor,
 /// - The class is generic.
 ///   In this case the class metadata was allocated from a generic
 ///   class metadata pattern by swift_allocateGenericClassMetadata().
-SWIFT_RUNTIME_EXPORT
-void swift_initClassMetadata(ClassMetadata *self,
-                             ClassLayoutFlags flags,
-                             size_t numFields,
-                             const TypeLayout * const *fieldTypes,
-                             size_t *fieldOffsets);
-
+///
+/// If HasObjCResilientClassStub is set in the flags, the last parameter
+/// must be specified. Swift will then call objc_realizeClassFromSwift()
+/// instead of objc_readClassPair(), passing in the stub as the first
+/// parameter.
+///
+/// Note: Swift 5.0 did not have an 'objcStub' parameter, so its value
+/// is not defined unless the flag is set.
```

Comment:
Is this the only reasonable way to do this?  It can't be linked from the type descriptor or something?

Reply:
It could be, yeah. I'll change it if you'd prefer to avoid a proliferation of new parameters to this function.

Reply:
It's code size for the caller and a more complex CC rule, so if it can be safely added to the TD, that would be better.  Will the stub reference only exist when deploying to newer runtimes?

Reply:
> Will the stub reference only exist when deploying to newer runtimes?

Correct.